src/test/scala/com/fulcrumgenomics/bam/ClipOverlappingReadsTest.scala

```diff
@@ -129,6 +127,21 @@ class ClipOverlappingReadsTest extends UnitSpec {
     r1.getAlignmentEnd >= r2.getAlignmentStart shouldBe false
   }
 
+  // This is a weird test that ensures that things terminate correctly when due to highly clipped
+  // input reads, one of the reads becomes unmapped during clipping
+  it should "handle reads that get unmapped because they are fully overlapped" in {
+    val builder = new SamRecordSetBuilder(readLength=50)
+    val clipper = new ClipOverlappingReads(input=dummyBam, output=dummyBam, ref=ref)
+    val Seq(r1a, r2a) = builder.addPair(name="q1", start1=100, start2=100, cigar1="3M47S", cigar2="48S2M")
+    val Seq(r1b, r2b) = builder.addPair(name="q2", start1=100, start2=100, cigar1="2M48S", cigar2="47S3M")
+
+    for ((r1, r2) <- Seq((r1a, r2a), (r1b, r2b))) {
+      clipper.clip(r1, r2)
+      val ok = r1.getReadUnmappedFlag || r2.getReadUnmappedFlag || r1.getAlignmentEnd < r2.getAlignmentStart
+      ok shouldBe true
+    }
```

Comment:
That's not actually the case though, and so I prefer the general test that they are either non-overlapping or at least one read got unmapped (hence not overlapping).

The current algorithm sizes the overlap (2 in this case) then attempts to clip half that from each read.  In this case, since the read with `3M` extends 1-base past the end of the other read, that's not sufficient.  So it again looks to clip 1 more base, and allocates the extra base to the R strand read, so ends up unmapping the R strand read in the first case (r1a, r2a), but simply shortening the R strand read in the second case.

I think it's better to test the desired outcome (either one gets unmapped or the overlap is clipped off) than testing specifics of the existing algorithm (which could potentially be improved).